R/zchunk_LA144.building_det_flsp.R

```diff
@@ -46,60 +48,379 @@ module_energy_LA144.building_det_flsp_DISABLED <- function(command, ...) {
+
+    # Reshape IEA data to long form
+    IEA_PCResFloorspace_long <- gather(IEA_PCResFloorspace, year, value_pcflsp, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year),
+             value_pcflsp = as.numeric(value_pcflsp))
+
+    # Create list of IEA iso's. It will be used to remove these iso's from Odyssee data.
+    # Also, create a list of years from IEA data in order to be used for filtering.
+    list_iso_IEA <- unique(IEA_PCResFloorspace_long$iso)
+    list_years_IEA <- unique(IEA_PCResFloorspace_long$year)
+
+    Odyssee_ResFloorspacePerHouse %>%
+      gather(year, value_phflsp, matches(YEAR_PATTERN)) %>% # Convert to long form
+      filter(!iso %in% list_iso_IEA) %>% # Remove iso's that are in IEA dataset
+      # left_join_error_no_match cannot be used because joining table does not contain every year, which will introduce NAs
+      left_join(A44.HouseholdSize_long, by = "year") %>%
+      mutate(year = as.integer(year)) %>% # Convert years to integer (or numeric) to be able to extrapolate
+      # Extrapolate, using rule 2 so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(value_pcdwelling = approx_fun(year, value_pcdwelling, rule = 2)) %>%
+      # Calculate per capita floorspace
+      mutate(value_pcflsp = value_phflsp / value_pcdwelling) %>%
+      select(iso, year, value_pcflsp) ->
+      L144.Odyssee_pcflsp_Yh
+
+    IEA_PCResFloorspace_long %>%
+      select(-country) %>% # Drop country column to bind with Odyssee table
+      bind_rows(L144.Odyssee_pcflsp_Yh) %>% # Bind with Odyssee table
+      filter(year %in% list_years_IEA) %>% # Restrict to year range from IEA table
+      # We want to drop any countries with all missing values. First we will drop all rows with missing per capita values,
+           # and then expand to all historical years. Countries with no per capita data for any year will consequently be removed.
+           # Note that this permanently removes Cyprus, which has data post 2004.
+      filter(!is.na(value_pcflsp)) %>%
+      group_by(iso) %>%
+      complete(year = HISTORICAL_YEARS) %>% # Exand table to all historical years.
+      ungroup() %>%
+      # Fill out missing values in specified countries
+      # USA data derived from RECS; see RGCAM data system for documentation
+      mutate(value_pcflsp = replace(value_pcflsp, iso == "usa" & year == "1980", 49.5)) %>%
+      spread(year, value_pcflsp) %>% # Spread data to be able to do inter-year calculations
+      # Missing 1990 values will be replaced by 1991, 1992, 1995, and 1996 (in that order)
+      mutate(`1990` = replace(`1990`, is.na(`1990`), `1991`[is.na(`1990`)]),
+             `1990` = replace(`1990`, is.na(`1990`), `1992`[is.na(`1990`)]),
+             `1990` = replace(`1990`, is.na(`1990`), `1995`[is.na(`1990`)]),
+             `1990` = replace(`1990`, is.na(`1990`), `1996`[is.na(`1990`)])) ->
```

Comment:
Hmm, I see what you mean. For now let's open an issue to come back to this and potentially change/fix later. Can you do so please?